### PR TITLE
CLN: fix bare excepts on close

### PR DESCRIPTION
**Pull request by jbrockmendel**
We're down to only a handful of bare `except`s.  This is the first of ~4 PRs to handle them all more carefully.

### CODE REVIEW

File: pandas/_libs/parsers.pyx [hunk 1]
```diff
@@ -566,11 +566,10 @@ cdef class TextReader:
     def close(self):
         # we need to properly close an open derived
         # filehandle here, e.g. and UTFRecoder
-        if self.handle is not None:
-            try:
-                self.handle.close()
-            except:
-                pass
+        if self.handle is not None and hasattr(self.handle, "close"):
```
Comment (jreback) [inline pandas/_libs/parsers.pyx]: yes I think we should just add it there

Comment (jreback) [reply pandas/_libs/parsers.pyx]: if you can update this here

Comment (jbrockmendel) [reply pandas/_libs/parsers.pyx]: changed

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**